### PR TITLE
Allow configurable compression level during build

### LINKED ISSUE #1522: Make compression level configurable
Our experiments Habitizing our Python monolith have been slowed down by very long build times for the `.hart` artifact - the compression step especially takes nearly 6 minutes. A method for speeding this up (lower compression option?) would be great.

@bixu works with us and says that Habitat us using `xz` internally, so maybe there could be a flag we can pass to `hab pkg build` to control the program's compression level?

cc @arochaf

### PR DESCRIPTION
**Pull request by jaym**
This patch adds a new flag to plan-build to allow setting the
compression level used by xz. This is helpful when using the
studio as a dev environment as it cuts off 10 seconds on a
simple package with just a go binary

To use in the studio:
```
build . -z 0
```

Fixes #1522

### CODE REVIEW

**Comment by thesentinels** [comment]
Thanks for the pull request! Here is what will happen next:
 1. Your PR will be reviewed by the maintainers
 2. If everything looks good, one of them will `approve` it, and your PR will be merged.

Thank you for contributing!

File: components/plan-build/bin/hab-plan-build.sh [hunk 1]
```diff
@@ -2006,7 +2008,7 @@ _generate_artifact() {
   mkdir -pv "$(dirname "$pkg_artifact")"
   rm -fv $tarf $xzf $pkg_artifact
   $_tar_cmd -cf $tarf $pkg_prefix
-  $_xz_cmd --compress -6 --threads=0 $tarf
+  $_xz_cmd --compress -${HART_COMPRESSION_LEVEL} --threads=0 $tarf
```
Comment (schisamo) [inline components/plan-build/bin/hab-plan-build.sh]: Wouldn't we want this ENV var to be prefixed with `HAB_` to match the idioms for other env vars?
https://www.habitat.sh/docs/reference/#environment-variables

Comment (jaym) [reply components/plan-build/bin/hab-plan-build.sh]: That makes sense. Are you ok with `HAB_HART_COMPRESSION_LEVEL` or just `HAB_COMPRESSION_LEVEL`. The latter seems not specific enough to me

Comment (schisamo) [reply components/plan-build/bin/hab-plan-build.sh]: Either seems fine by mean. Maybe someone from @habitat-sh/habitat-core-maintainers has an opinion or idiom that will make one naming approach more appealing.

Comment (raskchanky) [reply components/plan-build/bin/hab-plan-build.sh]: I agree with @jaym.  `HAB_HART_COMPRESSION_LEVEL` feels like the best choice there, at least to me.

**Comment by christophermaier** [comment]
@thesentinels approve

**Comment by thesentinels** [comment]
:metal: I am testing your branch against master before merging it. We do this to ensure that the master branch is never failing tests.

**Comment by thesentinels** [comment]
:neckbeard: Travis CI has [started testing this PR](https://travis-ci.org/habitat-sh/habitat/builds/339502612).

**Comment by thesentinels** [comment]
:sparkling_heart: Travis CI [reports this PR passed](https://travis-ci.org/habitat-sh/habitat/builds/339502612).

It always makes me feel nice when humans approve of one anothers work. I'm merging this PR now.

I just want you and the contributor to answer me one question:

gif-keyboard-3280869874741411265

**Pull request merged**